tasks/queue-granules/index.js

```diff
@@ -20,6 +25,30 @@ async function fetchGranuleProvider(prefix, providerId) {
   return JSON.parse(body);
 }
 
+/**
+ * Group granules by collection and split into batches then split again on provider
+ *
+ * @param {Array<Object>} granules - list of input granules
+ * @param {number} [batchSize] - size of batch of granules to queue
+ * @returns {Array<Object>} list of lists of granules: each list contains granules which belong
+ *                          to the same collection, and each list's max length is set by batchSize
+ */
+function groupAndBatchGranules(granules, batchSize) {
+  const filteredBatchSize = isNumber(batchSize) ? batchSize : 1;
+  const granulesByCollectionMap = groupBy(
```

Comment:
I assume we don't have concerns about the performance of this code itself for large numbers of granules (several thousand)?

Reply:
...No? 😄